applications/ShallowWaterApplication/python_scripts/visualization_mesh_process.py

```diff
@@ -1,39 +1,97 @@
+                "free_surface_variable"        : "FREE_SURFACE_ELEVATION",
+                "nodal_variables_to_transfer"  : [],
+                "nonhistorical_variables_to_transfer" : []
             }
             """
             )
         settings.ValidateAndAssignDefaults(default_settings)
 
-        self.process = KratosShallow.ShallowWaterVariablesUtility(Model[settings["model_part_name"].GetString()])
+        self.computing_model_part = Model[settings["model_part_name"].GetString()]
 
-        # The DefineDryProperties methods duplicates the current number of properties:
-        # For each property, it creates another one, which means dry stateself.
+        # Creating the utility for the topographic model part management
+        self.use_topographic_model_part = settings["create_topographic_model_part"].GetBool()
+        if self.use_topographic_model_part:
+            self.topographic_model_part = Model.CreateModelPart(settings["topographic_model_part_name"].GetString())
+            self.topography_utility = SW.ReplicateModelPartUtility(self.computing_model_part, self.topographic_model_part)
+            self.nodal_variables = self._GenerateVariableListFromInput(settings["nodal_variables_to_transfer"])
+            self.nonhistorical_variables = self._GenerateVariableListFromInput(settings["nonhistorical_variables_to_transfer"])
+            self.topography_variable = KM.KratosGlobals.GetVariable(settings["topography_variable"].GetString())
+            self.update_topography = settings["update_topography"].GetBool()
+            self.update_free_surface = settings["update_free_surface"].GetBool()
+
+        # The DefineAuxiliaryProperties method duplicates the current number of properties:
+        # For each property, it creates another one, which means dry state.
         # It should be called only once, otherwise, the number of properties will increase without limits
-        self.process.DefineDryProperties()
+        self.properties_utility = SW.PostProcessUtilities(self.computing_model_part)
+        self.properties_utility.DefineAuxiliaryProperties()
+
+    def ExecuteInitialize(self):
+        if self.use_topographic_model_part:
+            self.topography_utility.Replicate()
+
+    def ExecuteBeforeSolutionLoop(self):
+        if self.use_topographic_model_part:
+            for variable in self.nodal_variables:
+                self.topography_utility.TransferVariable(variable)
+            for variable in self.nonhistorical_variables:
+                self.topography_utility.TransferNonHistoricalVariable(variable)
+            self.topography_utility.SetDestinationMeshPosition(self.topography_variable)
+            if self.update_free_surface:
+                self.topography_utility.SetOriginMeshPosition(self.free_surface_variable)
+            else:
+                self.topography_utility.SetOriginMeshPosition()
+
+    def ExecuteInitializeSolutionStep(self):
```

Comment:
You can avoid this if you assign the same `ProcessInfo` container to the `topographic_model_part` when creating it.

Reply:
Thanks, very useful